### PR TITLE
WandbLogger improvement

### PR DESCRIPTION
**Pull request by elephantmipt**
## Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [x] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [x] Did you check the docs with `make check-docs`?
- [x] Did you write any new necessary tests?
- [x] Did you add your new functionality to the docs?
- [x] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/CHANGELOG.md)?
- [x] **You can use 'Login as guest' to see Teamcity build logs.**

## Related Issue

https://github.com/catalyst-team/catalyst/issues/782
Well, I spent a lot of time to figure out what's going wrong and came out with try except solution. I know that it is not super duper solution but why not?) It is better then change something with runner metrics key-value mapping.

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Now we can see beautiful pictures with gradients
picture_with_gradients
And with model structure
yet_another

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: catalyst/contrib/dl/callbacks/wandb_logger.py [hunk 1]
```diff
@@ -136,6 +136,7 @@ def key_locate(key: str):
     def on_stage_start(self, runner: IRunner):
         """Initialize Weights & Biases."""
         wandb.init(**self.logging_params, reinit=True, dir=str(runner.logdir))
+        wandb.watch(runner.model)
```
Comment (Scitator) [inline catalyst/contrib/dl/callbacks/wandb_logger.py]: could you please parameterize this?

File: catalyst/contrib/dl/callbacks/wandb_logger.py [hunk 2]
```diff
@@ -177,13 +178,22 @@ def on_epoch_end(self, runner: IRunner):
...
+                self._log_metrics(
+                    metrics=splitted_epoch_metrics[extra_mode],
+                    step=runner.global_epoch,
+                    mode=extra_mode,
+                    suffix=self.epoch_log_suffix,
+                    commit=True,
+                )
+            except KeyError:
+                self._log_metrics(
+                    metrics=splitted_epoch_metrics,
```
Comment (Scitator) [inline catalyst/contrib/dl/callbacks/wandb_logger.py]: there is the difference between params, `splitted_epoch_metrics[extra_mode]` vs `splitted_epoch_metrics`. why so?

File: catalyst/contrib/dl/callbacks/wandb_logger.py [hunk 3]
```diff
@@ -177,13 +178,22 @@ def on_epoch_end(self, runner: IRunner):
...
-            )
+            try:
+                self._log_metrics(
+                    metrics=splitted_epoch_metrics[extra_mode],
+                    step=runner.global_epoch,
+                    mode=extra_mode,
+                    suffix=self.epoch_log_suffix,
+                    commit=True,
+                )
+            except KeyError:
```
Comment (Scitator) [inline catalyst/contrib/dl/callbacks/wandb_logger.py]: meybe, it's better to use `if key in .keys():`?

File: catalyst/contrib/dl/callbacks/wandb_logger.py (hunk 2 above)
Comment (elephantmipt) [reply catalyst/contrib/dl/callbacks/wandb_logger.py]: I attached issue to this PR. Sometimes I get an KeyError both in config API and in notebook API. Here is an example. Just try to run it
```python
import os
import torch
from torch.nn import functional as F
from torch.utils.data import DataLoader
from catalyst import dl
from catalyst.contrib.data.transforms import ToTensor
from catalyst.contrib.datasets import MNIST
from catalyst.utils import metrics
from catalyst.contrib.dl import WandbLogger

model = torch.nn.Linear(28 * 28, 10)
optimizer = torch.optim.Adam(model.parameters(), lr=0.02)

loaders = {
    "train": DataLoader(MNIST(os.getcwd(), train=True, download=True, transform=ToTensor()), batch_size=32),
    "valid": DataLoader(MNIST(os.getcwd(), train=False, download=True, transform=ToTensor()), batch_size=32),
}

class CustomRunner(dl.Runner):

    def predict_batch(self, batch):
        # model inference step
        return self.model(batch[0].to(self.device).view(batch[0].size(0), -1))

    def _handle_batch(self, batch):
        # model train/valid step
        x, y = batch
        y_hat = self.model(x.view(x.size(0), -1))

        loss = F.cross_entropy(y_hat, y)
        accuracy01, accuracy03 = metrics.accuracy(y_hat, y, topk=(1, 3))
        self.batch_metrics.update(
            {"loss": loss, "accuracy01": accuracy01, "accuracy03": accuracy03}
        )

        if self.is_train_loader:
            loss.backward()
            self.optimizer.step()
            self.optimizer.zero_grad()

runner = CustomRunner()
# model training
runner.train(
    model=model,
    optimizer=optimizer,
    loaders=loaders,
    logdir="./logs",
    num_epochs=5,
    verbose=True,
    load_best_on_end=True,
    callbacks={
        "logger": WandbLogger(
        )
    },
    timeit=True
)
```

File: catalyst/contrib/dl/callbacks/wandb_logger.py [hunk 4]
```diff
@@ -63,6 +63,7 @@ def __init__(
         metric_names: List[str] = None,
         log_on_batch_end: bool = False,
         log_on_epoch_end: bool = True,
+        log: str = "gradients",
```
Comment (Scitator) [inline catalyst/contrib/dl/callbacks/wandb_logger.py]: is it possible to make it `None` by default?

**Review by Scitator — DISMISSED** [review]
Thanks for PR.
Should be automerged after https://github.com/catalyst-team/catalyst/pull/837 codestyle and docs update :)
